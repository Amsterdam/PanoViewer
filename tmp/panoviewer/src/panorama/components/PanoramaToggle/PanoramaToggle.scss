@import "../../../shared/styles/config";

.c-panorama-toggle {
  @include font-size($s-font);
  position: relative;
}

.c-panorama-toggle__button {
  @extend %hover-focus-button;
  @include font-size($s-font);
  @include font-weight($bold-weight);

  background-color: $primary-light-alpha48;
  border: 1px solid $secondary-gray20;
  cursor: pointer;
  height: $button-size;
  line-height: #{$button-size - 3};
  min-width: 202px;
  padding: 0 #{$button-size + 6};

  &::before {
    @extend %icon-button-text-extra;
    @include sprite("clock-black");

    left: 4px;
    position: absolute;
    top: 0;
  }

  &::after {
    @extend %icon-button-text-extra;
    @include sprite("up-black");

    position: absolute;
    right: 4px;
    top: 0;
  }

  &.c-panorama-toggle__button--active {
    @extend %active-button;

    &::after {
      @include sprite("down-black");
    }
  }
}

.c-panorama-toggle__menu {
  background-color: $primary-light;
  border: 1px solid $secondary-gray20;
  bottom: $button-size;
  box-shadow: 2px 0 0 0 $secondary-gray40;
  padding: 12px 0 6px;
  position: absolute;
}

.c-panorama-toggle__item {
  @extend %hover-focus-listitem;

  cursor: pointer;
  font-family: "Avenir LT W01 85 Heavy";
  height: 32px;
  line-height: 32px;
  padding: 0 32px;
  text-align: left;
  white-space: nowrap;
  width: 100%;

  &.c-panorama-toggle__item--active {
    background-color: $secondary-gray10;

    &:hover,
    &:active,
    &:focus {
      background-color: $secondary-gray4 !important;
    }
  }

  &:hover,
  &:active,
  &:focus {
    background-color: $secondary-gray4 !important;
  }
}

.c-panorama-toggle__devider {
  background-color: $secondary-gray20;
  height: 1px;
  margin: 12px 0 6px;
}

.c-panorama-toggle__external {
  height: 32px;
  padding: 0 20px 0 16px;
  white-space: nowrap;
}

.c-panorama-toggle__external-link {
  line-height: 32px;
  opacity: .6;

  &:active,
  &:hover,
  &:focus {
    opacity: 1;
  }
}
