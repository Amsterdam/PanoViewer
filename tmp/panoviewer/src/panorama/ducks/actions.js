// Actions creators
import {
  CLOSE_PANORAMA,
  FETCH_PANORAMA_ERROR,
  FETCH_PANORAMA_HOTSPOT_REQUEST,
  FETCH_PANORAMA_REQUEST,
  FETCH_PANORAMA_REQUEST_EXTERNAL,
  FETCH_PANORAMA_SUCCESS,
  SET_PANORAMA_LOCATION,
  SET_PANORAMA_ORIENTATION,
  SET_PANORAMA_TAGS
} from './constants';

export const fetchPanoramaRequest = (payload) => ({
  type: FETCH_PANORAMA_REQUEST,
  payload,
  meta: {
    tracking: true
  }
});

export const fetchPanoramaHotspotRequest = (payload) => ({
  type: FETCH_PANORAMA_HOTSPOT_REQUEST,
  payload,
  meta: {
    tracking: true
  }
});

export const setPanoramaTags = (payload) => ({
  type: SET_PANORAMA_TAGS,
  payload,
  meta: {
    tracking: payload
  }
});

export const fetchPanoramaRequestExternal = () => ({
  type: FETCH_PANORAMA_REQUEST_EXTERNAL,
  meta: {
    tracking: true
  }
});

export const fetchPanoramaSuccess = (payload) => ({
  type: FETCH_PANORAMA_SUCCESS,
  payload,
  meta: {
    tracking: payload
  }
});
export const fetchPanoramaError = (error) => ({
  type: FETCH_PANORAMA_ERROR,
  payload: error
});

export const closePanorama = () => ({
  type: CLOSE_PANORAMA
});
export const setPanoramaLocation = (payload) => ({
  type: SET_PANORAMA_LOCATION,
  payload,
  meta: {
    tracking: true
  }
});
export const setPanoramaOrientation = ({ heading, pitch, fov }) => ({
  type: SET_PANORAMA_ORIENTATION,
  payload: {
    heading,
    pitch,
    fov
  }
});
