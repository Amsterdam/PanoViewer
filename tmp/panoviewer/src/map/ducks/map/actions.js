import { getLabelObjectByTags } from '../../../panorama/ducks/selectors';
import { normalizeCoordinate } from '../../../shared/services/coordinate-reference-system';
import {
  MAP_EMPTY_GEOMETRY, MAP_UPDATE_SHAPE, MAP_SET_DRAWING_MODE, MAP_END_DRAWING,
  MAP_CLEAR, MAP_ZOOM, TOGGLE_MAP_PANEL, CLOSE_MAP_PANEL, SET_MAP_BASE_LAYER,
  TOGGLE_MAP_OVERLAY, TOGGLE_MAP_OVERLAY_PANORAMA, TOGGLE_MAP_OVERLAY_VISIBILITY,
  MAP_PAN, SET_MAP_CLICK_LOCATION, MAP_BOUNDING_BOX, MAP_LOADING
} from './constants';

// Actions
export const mapEmptyGeometry = () => ({ type: MAP_EMPTY_GEOMETRY });
export const mapUpdateShape = (payload) => ({ type: MAP_UPDATE_SHAPE, payload });
export const mapSetDrawingMode = (payload) => ({
  type: MAP_SET_DRAWING_MODE,
  payload,
  meta: {
    tracking: payload.drawingMode
  }
});
export const mapEndDrawing = (payload) => ({ type: MAP_END_DRAWING, payload });
export const mapClear = () => ({ type: MAP_CLEAR });
export const updateZoom = (payload) => ({ type: MAP_ZOOM, payload });
export const toggleMapPanel = () => ({ type: TOGGLE_MAP_PANEL });
export const closeMapPanel = () => ({ type: CLOSE_MAP_PANEL });
export const setMapBaseLayer = (payload) => ({
  type: SET_MAP_BASE_LAYER,
  payload,
  meta: {
    tracking: payload
  }
});
export const toggleMapOverlay = (payload) => ({
  type: TOGGLE_MAP_OVERLAY,
  payload: {
    mapLayers: (payload.id) ? [payload.id] : payload.legendItems.map((overlay) => overlay.id)
  },
  meta: {
    tracking: payload
  }
});

export const toggleMapOverlayPanorama = (tags) => ({
  type: TOGGLE_MAP_OVERLAY_PANORAMA,
  payload: getLabelObjectByTags(tags).layerId
});

export const toggleMapOverlayVisibility = (mapLayerId, isVisible) => ({
  type: TOGGLE_MAP_OVERLAY_VISIBILITY,
  mapLayerId,
  isVisible: !isVisible
});
export const updatePan = (payload) => ({
  type: MAP_PAN,
  payload: {
    latitude: normalizeCoordinate(payload.lat, 7),
    longitude: normalizeCoordinate(payload.lng, 7)
  }
});
export const setSelectedLocation = (payload) => ({
  type: SET_MAP_CLICK_LOCATION,
  payload: {
    location: {
      latitude: normalizeCoordinate(payload.latlng.lat, 7),
      longitude: normalizeCoordinate(payload.latlng.lng, 7)
    }
  },
  meta: {
    tracking: true
  }
});
export const updateBoundingBox = (payload) => ({
  type: MAP_BOUNDING_BOX,
  payload
});

export const mapLoadingAction = (payload) => ({ type: MAP_LOADING, payload });
