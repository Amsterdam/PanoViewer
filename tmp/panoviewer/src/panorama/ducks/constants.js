import { toHome } from '../../store/redux-first-router/actions';

export const REDUCER_KEY = 'panorama';
export const FETCH_PANORAMA_REQUEST = `${REDUCER_KEY}/FETCH_PANORAMA_REQUEST`;
export const FETCH_PANORAMA_SUCCESS = `${REDUCER_KEY}/FETCH_PANORAMA_SUCCESS`;
export const FETCH_PANORAMA_ERROR = `${REDUCER_KEY}/FETCH_PANORAMA_ERROR`;
export const SET_PANORAMA_ORIENTATION = `${REDUCER_KEY}/SET_PANORAMA_ORIENTATION`;
export const SET_PANORAMA_LOCATION = `${REDUCER_KEY}/SET_PANORAMA_LOCATION`;
export const CLOSE_PANORAMA = `${REDUCER_KEY}/CLOSE_PANORAMA`;
export const FETCH_PANORAMA_HOTSPOT_REQUEST = `${REDUCER_KEY}/FETCH_PANORAMA_HOTSPOT_REQUEST`;
export const SET_PANORAMA_TAGS = `${REDUCER_KEY}/FETCH_PANORAMA_REQUEST_TOGGLE`;
export const FETCH_PANORAMA_REQUEST_EXTERNAL = `${REDUCER_KEY}/FETCH_PANORAMA_REQUEST_EXTERNAL`;

export const PANO_LABELS = [
  {
    layerId: 'pano',
    label: 'Meest recent',
    tags: ['mission-bi']
  },
  {
    layerId: 'pano2018bi',
    label: 'Alleen 2018',
    tags: ['mission-bi', 'mission-2018']
  },
  {
    layerId: 'pano2017bi',
    label: 'Alleen 2017',
    tags: ['mission-bi', 'mission-2017']
  },
  {
    layerId: 'pano2016bi',
    label: 'Alleen 2016',
    tags: ['mission-bi', 'mission-2016']
  },
  {
    layerId: 'pano2018woz',
    label: 'Alleen 2018 WOZ',
    tags: ['mission-woz', 'mission-2018']
  },
  {
    layerId: 'pano2017woz',
    label: 'Alleen 2017 WOZ',
    tags: ['mission-woz', 'mission-2017']
  }
];

export const initialState = {
  location: null,   // eg: [52.8, 4.9]
  tags: PANO_LABELS[0].tags,
  targetLocation: null,
  pitch: 0,         // eg: -10
  heading: 0,       // eg: 270
  fov: null,        // eg: 65
  image: null,      // eg: {
                    //     pattern: 'http://www.example.com/path/some-id/{this}/{that}/{thingie}.jpg',
                    //     preview: 'http://www.example.com/path/some-id/preview.jpg'
                    // }
  hotspots: [],     // eg: [{id: 'ABC124', heading: 90, distance: 18}],
  date: null,       // eg: new Date()
  isLoading: true,
  detailReference: [],
  pageReference: ''
};

export const PAGE_REFS = {
  HOME: 'home'
};

export const PAGE_REF_MAPPING = {
  [PAGE_REFS.HOME]: toHome
};
