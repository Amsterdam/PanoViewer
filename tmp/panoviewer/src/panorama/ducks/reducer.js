import { PANORAMA_CONFIG } from '../services/panorama-api/panorama-api';
import PAGES from '../../app/pages';
import {
  FETCH_PANORAMA_ERROR,
  FETCH_PANORAMA_HOTSPOT_REQUEST,
  FETCH_PANORAMA_REQUEST,
  FETCH_PANORAMA_SUCCESS,
  initialState,
  REDUCER_KEY,
  SET_PANORAMA_LOCATION,
  SET_PANORAMA_ORIENTATION,
  SET_PANORAMA_TAGS
} from './constants';
import paramsRegistry from '../../store/params-registry';
import { shouldResetState } from '../../store/redux-first-router/actions';

export { REDUCER_KEY as PANORAMA };

export default function reducer(state = initialState, action) {
  if (shouldResetState(action, [PAGES.PANORAMA])) {
    return initialState;
  }

  const enrichedState = {
    ...state,
    ...paramsRegistry.getStateFromQueries(REDUCER_KEY, action)
  };

  switch (action.type) {
    case FETCH_PANORAMA_HOTSPOT_REQUEST:
    case FETCH_PANORAMA_REQUEST:
      return {
        ...enrichedState,
        isLoading: true,
        targetLocation: null
      };

    case SET_PANORAMA_TAGS:
      return {
        ...enrichedState,
        tags: action.payload
      };

    case FETCH_PANORAMA_SUCCESS:
      return {
        ...enrichedState,
        id: action.payload.id,
        date: action.payload.date,
        pitch: enrichedState.pitch || initialState.pitch,
        fov: enrichedState.fov || PANORAMA_CONFIG.DEFAULT_FOV,
        hotspots: action.payload.hotspots,
        isLoading: false,
        location: action.payload.location,
        image: action.payload.image
      };
    case FETCH_PANORAMA_ERROR:
      return {
        ...enrichedState,
        isLoading: false,
        isError: true // TODO: refactor, show error
      };

    case SET_PANORAMA_ORIENTATION:
      return {
        ...enrichedState,
        heading: action.payload.heading,
        pitch: action.payload.pitch,
        fov: action.payload.fov
      };

    case SET_PANORAMA_LOCATION:
      return {
        ...enrichedState,
        location: action.payload,
        targetLocation: action.payload,
        detailReference: initialState.detailReference
      };

    default:
      return enrichedState;
  }
}

