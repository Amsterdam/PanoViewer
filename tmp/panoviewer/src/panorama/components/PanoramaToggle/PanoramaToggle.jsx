import React from 'react';
import PropTypes from 'prop-types';

import './PanoramaToggle.scss';
import { PANO_LABELS } from '../../ducks/constants';

const getStreetViewUrl = (location, heading) => {
  const [latitude, longitude] = location;
  const path = 'http://maps.google.com/maps?q=&layer=c&';
  const parameters = `cbll=${latitude},${longitude}&cbp=11,${heading},0,0,0`;

  return `${path}${parameters}`;
};

class PanoramaToggle extends React.Component {
  constructor() {
    super();
    this.state = {
      showMenu: false
    };
    this.toggleMenu = this.toggleMenu.bind(this);
    this.setPanoramaTags = this.setPanoramaTags.bind(this);
    this.onOpenPanoramaExternal = this.onOpenPanoramaExternal.bind(this);
  }

  onOpenPanoramaExternal() {
    this.props.openPanoramaExternal();
    this.toggleMenu();
  }

  setPanoramaTags(option) {
    this.props.setPanoramaTags(option);
    this.toggleMenu();
  }

  toggleMenu() {
    this.setState({ showMenu: !this.state.showMenu });
  }

  render() {
    const { heading, currentLabel, location } = this.props;
    const { showMenu } = this.state;

    return (
      <div className="c-panorama-status-bar__history">
        <div className="c-panorama-toggle qa-panorama-toggle">
          <button
            className={`
              c-panorama-toggle__button
              qa-panorama-toggle__button
              ${(showMenu) ? 'c-panorama-toggle__button--active' : ''}
            `}
            onClick={this.toggleMenu}
          >
            {currentLabel}
          </button>
          {(showMenu) ? <ul className="c-panorama-toggle__menu qa-panorama-toggle__menu">
            {PANO_LABELS.map((label) => (
              <li
                key={label.layerId}
              >
                <button
                  className={`
                    c-panorama-toggle__item
                    qa-panorama-toggle__item
                  `}
                  onClick={() => this.setPanoramaTags(label.tags)}
                >
                  {label.label}
                </button>
              </li>
            ))}
            <li
              key="divider"
              className="c-panorama-toggle__devider qa-panorama-toggle__devider"
            />
            <li
              key="google-street-view"
              className="c-panorama-toggle__external qa-panorama-toggle__external"
            >
              <a
                href={getStreetViewUrl(location, heading)}
                target="_blank"
                rel="noopener"
                className="
                  c-link
                  c-link--arrow
                  c-panorama-toggle__external-link
                  qa-panorama-toggle__external-link"
                onClick={this.onOpenPanoramaExternal}
              >Google Street View</a>
            </li>
          </ul> : ''}
        </div>
      </div>
    );
  }
}

PanoramaToggle.defaultProps = {
  fetchPanoramaRequest: ''
};

PanoramaToggle.propTypes = {
  heading: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  currentLabel: PropTypes.string.isRequired,
  location: PropTypes.array.isRequired,  // eslint-disable-line
  setPanoramaTags: PropTypes.oneOfType([PropTypes.string, PropTypes.func]), // eslint-disable-line
  openPanoramaExternal: PropTypes.oneOfType([PropTypes.string, PropTypes.func]) // eslint-disable-line
};

export default PanoramaToggle;
