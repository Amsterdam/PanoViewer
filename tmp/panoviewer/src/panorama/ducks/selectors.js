import { createSelector } from 'reselect';
import {
  panoramaOrientationType,
  panoramaPersonType
} from '../../map/components/leaflet/services/icons.constant';
import { PANO_LABELS, REDUCER_KEY } from './constants';

export const getPanorama = (state) => state[REDUCER_KEY] || {};
export const getPanoramaLocation = createSelector(
  getPanorama,
  (panorama) => (
    panorama ? panorama.location : []
  )
);
const getStateOfKey = (key) => (state) => createSelector(getPanorama, (data) => (data[key]))(state);

export const getPanoramaHeading = getStateOfKey('heading');
export const getPanoramaMarkers = createSelector([getPanoramaLocation, getPanoramaHeading],
  (location, heading) => (
    location ? [
      {
        position: location,
        type: panoramaOrientationType,
        heading: heading || 0
      },
      {
        position: location,
        type: panoramaPersonType
      }
    ] : []
  )
);

export const getPanoramaPitch = getStateOfKey('pitch');
export const getDetailReference = getStateOfKey('detailReference');
export const getPageReference = getStateOfKey('pageReference');
export const getPanoramaTags = getStateOfKey('tags');

export const getLabelObjectByTags = (tags) =>
  PANO_LABELS.find((value) => (
    JSON.stringify(value.tags.sort()) === JSON.stringify(tags)
  )) || PANO_LABELS[0];
