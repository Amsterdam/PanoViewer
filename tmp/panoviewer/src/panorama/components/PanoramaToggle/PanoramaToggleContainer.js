import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { fetchPanoramaRequestExternal, setPanoramaTags } from '../../ducks/actions';
import PanoramaToggle from './PanoramaToggle';

const mapStateToProps = () => ({});

const mapDispatchToProps = (dispatch) => bindActionCreators({
  setPanoramaTags,
  openPanoramaExternal: fetchPanoramaRequestExternal
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(PanoramaToggle);
