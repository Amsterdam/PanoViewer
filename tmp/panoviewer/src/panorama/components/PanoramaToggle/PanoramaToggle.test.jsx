import React from 'react';
import { shallow } from 'enzyme';

import { setPanoramaTags } from '../../ducks/actions';

import PanoramaToggle from './PanoramaToggle';

jest.mock('../../ducks/actions');

describe('PanoramaToggle', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <PanoramaToggle
        {...{
          heading: 999,
          currentLabel: 'Meest recent',
          location: [2, 3],
          setPanoramaTags: jest.fn
        }}
      />
    );
  });

  setPanoramaTags.mockReturnValue({ type: '' });

  it('should render everything', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('menu opens on button click', () => {
    expect(wrapper.instance().state.showMenu).toBe(false);

    wrapper.find('.c-panorama-toggle__button').simulate('click');

    expect(wrapper.instance().state.showMenu).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });

  it('fires panoramaRequest on button click in menu', () => {
    wrapper.instance().setState({ showMenu: true });
    wrapper.update();

    wrapper.find('.c-panorama-toggle__item').at(1).simulate('click');
    expect(wrapper.instance().state.showMenu).toBe(false);
  });
});
