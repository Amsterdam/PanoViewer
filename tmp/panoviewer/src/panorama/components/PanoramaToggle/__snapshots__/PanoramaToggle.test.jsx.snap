// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanoramaToggle menu opens on button click 1`] = `
<div
  className="c-panorama-status-bar__history"
>
  <div
    className="c-panorama-toggle qa-panorama-toggle"
  >
    <button
      className="
              c-panorama-toggle__button
              qa-panorama-toggle__button
              c-panorama-toggle__button--active
            "
      onClick={[Function]}
    >
      Meest recent
    </button>
    <ul
      className="c-panorama-toggle__menu qa-panorama-toggle__menu"
    >
      <li
        key="pano"
      >
        <button
          className="
                    c-panorama-toggle__item
                    qa-panorama-toggle__item
                  "
          onClick={[Function]}
        >
          Meest recent
        </button>
      </li>
      <li
        key="pano2018bi"
      >
        <button
          className="
                    c-panorama-toggle__item
                    qa-panorama-toggle__item
                  "
          onClick={[Function]}
        >
          Alleen 2018
        </button>
      </li>
      <li
        key="pano2017bi"
      >
        <button
          className="
                    c-panorama-toggle__item
                    qa-panorama-toggle__item
                  "
          onClick={[Function]}
        >
          Alleen 2017
        </button>
      </li>
      <li
        key="pano2016bi"
      >
        <button
          className="
                    c-panorama-toggle__item
                    qa-panorama-toggle__item
                  "
          onClick={[Function]}
        >
          Alleen 2016
        </button>
      </li>
      <li
        key="pano2018woz"
      >
        <button
          className="
                    c-panorama-toggle__item
                    qa-panorama-toggle__item
                  "
          onClick={[Function]}
        >
          Alleen 2018 WOZ
        </button>
      </li>
      <li
        key="pano2017woz"
      >
        <button
          className="
                    c-panorama-toggle__item
                    qa-panorama-toggle__item
                  "
          onClick={[Function]}
        >
          Alleen 2017 WOZ
        </button>
      </li>
      <li
        className="c-panorama-toggle__devider qa-panorama-toggle__devider"
        key="divider"
      />
      <li
        className="c-panorama-toggle__external qa-panorama-toggle__external"
        key="google-street-view"
      >
        <a
          className=" c-link c-link--arrow c-panorama-toggle__external-link qa-panorama-toggle__external-link"
          href="http://maps.google.com/maps?q=&layer=c&cbll=2,3&cbp=11,999,0,0,0"
          onClick={[Function]}
          rel="noopener"
          target="_blank"
        >
          Google Street View
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`PanoramaToggle should render everything 1`] = `
<div
  className="c-panorama-status-bar__history"
>
  <div
    className="c-panorama-toggle qa-panorama-toggle"
  >
    <button
      className="
              c-panorama-toggle__button
              qa-panorama-toggle__button
              
            "
      onClick={[Function]}
    >
      Meest recent
    </button>
  </div>
</div>
`;
